cms/djangoapps/contentstore/views/course.py

```diff
@@ -985,11 +986,12 @@ def settings_handler(request, course_key_string):
                 settings.FEATURES.get('EDITABLE_SHORT_DESCRIPTION', True)
             )
             self_paced_enabled = SelfPacedConfiguration.current().enabled
+            course_overview = CourseOverview.get_from_id(course_key)
```

Comment:
To date, the CMS hasn't used `CourseOverview` as `CourseOverview` is meant to be an LMS cache of Course metadata.
Also, to date, we've allowed most low-level functionality to except either a `course_module` object (from the Modulestore) or a `CourseOverview` object from SQL.  
So, may I suggest that you update `get_link_for_about_page` to accept either course_overview or course_module.  And then use `getattr` to retrieve the marketing_url.